frontend/public/components/monitoring/query-browser.tsx

```diff
@@ -501,6 +501,9 @@ const QueryBrowser_: React.FC<QueryBrowserProps> = ({
   tickInterval,
   timespan,
 }) => {
+  // Min height for GraphEmpty(empty state)
+  const EMPTY_GRAPH_MIN_HEIGHT = 310;
```

Comment:
Do we need this? I think the height should still be reasonable without it. We already have a state where an error message is displayed without a graph and I think the height is OK in that case without forcing a `height`.

Is there a page that looks bad without `EMPTY_GRAPH_MIN_HEIGHT`?

Also, I notice that `310` is too much for some places, like the Admin perspective > Monitoring > Metrics page.

Ideally, we wouldn't use `EMPTY_GRAPH_MIN_HEIGHT` and just fallback to CSS to give a reasonable height.

Reply:
`QueryBrowser` already has a CSS class `graph-empty-state` which sets min height `310px` for empty state
but the `GraphEmpty` defaults to `180px` which made it seem floating towards the top.

**Topology Sidebar**
![Screenshot from 2020-08-28 17-37-48](https://user-images.githubusercontent.com/38663217/91560022-bd438700-e956-11ea-8f99-0104b08ec885.png)

**Dev perspective -> Monitoring**
![Screenshot from 2020-08-28 17-38-21](https://user-images.githubusercontent.com/38663217/91560095-de0bdc80-e956-11ea-8558-20d276feca57.png)

Reply:
One problem with this is that depending on the page, the `graph-empty-state` wrapper may contain other content in addition to the empty state message (like the graph controls on the Metrics page), so the `310px` actually makes the `graph-empty-state` taller than 310 pixels.

Not sure if there is an easy CSS fix here.

Reply:
If we don't want to set the height then we can do it by using `display: table` CSS property. WDYT @kyoto?

Reply:
^ @christianvogt What will be the better way to fix this?

Reply:
The `display: table` table approach feels like a bit of a hack to me, but not sure what a cleaner solution would be? 🤔

Reply:
Apply `display: flex` to `graph-empty-state` ?

Reply:
Yes, it works. This was the only thing I didn't try. Thanks @christianvogt

Reply:
tried it out. The empty state message is aligned correctly. updated the code
Thanks @vikram-raj @christianvogt

Reply:
Unfortunately this introduces a bug where the graph controls are now vertically centered while the graph is loading.

![screenshot](https://user-images.githubusercontent.com/460802/92668898-029a8800-f34b-11ea-9f70-3f93932bf855.png)

Reply:
Looks like the styling needs to be applied more conditionally since it's only necessary when the empty state is shown

Reply:
added conditional styling
![Screenshot from 2020-09-15 02-17-44](https://user-images.githubusercontent.com/38663217/93136533-b90abe00-f6f9-11ea-9962-c61f934d75fc.png)

Reply:
Unfortunately, I think this is introducing a different bug on the Metrics page. When the graph is loading, the graph wrapper loses its height, which I think is not what we want since it causes everything below the graph to jump up / down as the graph is loading.

![screenshot](https://user-images.githubusercontent.com/460802/93150904-546d5480-f735-11ea-96d4-97791d7c947d.png)

Reply:
I've pushed a fix addressing the css issue. Please take a look